go/vt/key/key.go

```diff
@@ -296,3 +299,52 @@ func ParseShardingSpec(spec string) ([]*topodatapb.KeyRange, error) {
 	}
 	return ranges, nil
 }
+
+// ParseDestination parses the string representation of a Destionation
+// of the form keyspace:shard@tablet_type. You can use a / instead of a :.
+func ParseDestination(targetString string, defaultTabletType topodatapb.TabletType) (Destination, string, topodatapb.TabletType, error) {
+	var dest Destination
+	var keyspace string
+	tabletType := defaultTabletType
+
+	last := strings.LastIndexAny(targetString, "@")
+	if last != -1 {
+		tabletType = parseTabletType(targetString[last+1:])
+		targetString = targetString[:last]
+	}
+	last = strings.LastIndexAny(targetString, "/:")
+	if last != -1 {
+		dest = DestinationShard(targetString[last+1:])
+		targetString = targetString[:last]
+	}
+	// Try to parse it as a range
+	last = strings.LastIndexAny(targetString, "[")
+	if last != -1 {
+		rangeEnd := strings.LastIndexAny(targetString, "]")
+		if rangeEnd == -1 {
+			return dest, keyspace, tabletType, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "invalid key range provided. Couldn't find range end ']'")
+
+		}
+		rangeString := targetString[last+1 : rangeEnd]
+		keyRange, err := ParseShardingSpec(rangeString)
+		if err != nil {
+			return dest, keyspace, tabletType, err
+		}
+		if len(keyRange) != 1 {
+			return dest, keyspace, tabletType, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "single keyrange expected in %s", rangeString)
+		}
+		dest = DestinationExactKeyRange{KeyRange: keyRange[0]}
+		targetString = targetString[:last]
+	}
+	keyspace = targetString
+	return dest, keyspace, tabletType, nil
+}
+
+// ParseTabletType parses the tablet type into the enum.
+func parseTabletType(param string) topodatapb.TabletType {
+	value, ok := topodatapb.TabletType_value[strings.ToUpper(param)]
+	if !ok {
+		return topodatapb.TabletType_UNKNOWN
```

Comment:
I think this method should return an error:
func parseTabletType(param string) (topodatapb.TabletType, error) {

Reply:
After I moved ParseDestination out of key.go, I don't longer need this method and I'm using `topoproto.ParseTabletType`